### ISSUE TITLE
Task.sleep is extremely slow on Windows (100-200x slower than macOS/Linux)

### ISSUE THREAD

**Issue by jakepetroules** [swiftlang/swift#75945]
### Description

Task.sleep appears to be significantly slower than expected on Windows.

### Reproduction

```swift
import Foundation

let clock = ContinuousClock()
let duration = try await clock.measure {
    for _ in 0..<100 {
        try await Task.sleep(for: .microseconds(1000))
    }
}

print(duration)
```

prints 26.889839124999998 seconds on Windows

### Expected behavior

Linux: prints 0.128345333 seconds

```
$ swiftc --version
Swift version 5.10.1 (swift-5.10.1-RELEASE)
Target: aarch64-unknown-linux-gnu

$ cat /etc/debian_version
12.6

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 12 (bookworm)
Release:	12
Codename:	bookworm
```

macOS: prints 0.201219667 seconds

--

I'd expect Windows to perform comparably, rather than be off by roughly 100x-200x.

### Environment

swiftc --version
Swift version 6.0-dev (LLVM 31f0cdc6d5ff8a6, Swift aa0786cc7b3190c)
Target: x86_64-unknown-windows-msvc
Microsoft Windows [Version 10.0.22631.4037]
Parallels ARM virtual machine

### Additional information

Removing the for loop and changing the duration to 100,000 microseconds results in printing almost exactly 0.1 seconds on macOS and Linux (as expected), but 0.385826416 seconds on Windows (almost 4x as long).

**Comment by HeMet** [comment]
Maybe it is the same issue as https://github.com/swiftlang/swift/issues/72095

**Comment by al45tair** [comment]
While we've closed this one in favour of #72095, sleeping for 1ms, which is what this is doing, appears to take on average 15ms when using the method Dispatch is using to do the sleep. Doing that 100 times would therefore take at least 1.5s, which is maybe not 26s, but it's still a lot longer than you'd expect.

I can believe, given the complicated way the timers are implemented on Windows, that there could be more pathological cases where things are even worse.

**Comment by al45tair** [comment]
Just FYI, on my VM, I get `6.56` seconds when I run this program.
